src/exchangeRateRouter.ts

```diff
@@ -26,6 +26,16 @@ const asExchangeRatesReq = asObject({
 })
 
 export type ExchangeRateReq = ReturnType<typeof asExchangeRateReq>
+export type ExchangeRatesReq = ReturnType<typeof asExchangeRatesReq>
+
+declare module 'express' {
+  interface Request {
+    requestedRates: ExchangeRatesReq
+    requestedRatesResult: ReturnGetRate
+  }
+}
+
+type Req = express.Request
```

Comment:
Ditch this for a helper cleaner:

**asExtendedReq.ts**

```ts
export const asExtendedReq = <T>(asExtension: Cleaner<T>) => (
  raw: any
): express.Request & T => {
  const extension = asExtension(raw)
  return {
    ...extension,
    ...raw
  }
}
```

**excahngeRateRouter.ts**

```ts
const asRatesRequest = asExtendedReq(
  asObject({
    requestedRates: asExchangeRatesReq
  })
)

// ... 
 
const exReq = asMaybe(asExtendedReq)(req)

if (exReq == null) return next(500) // respond 500
```